provisioner_play.go

```diff
@@ -140,17 +140,41 @@ func (r *runnablePlay) ToLocalCommand(o terraform.UIOutput, rpla runnablePlayLoc
 }
 
 type runnablePlayLocalAnsibleArgs struct {
-	Username       string
-	Port           int
-	PemFile        string
-	KnownHostsFile string
+	Username        string
+	Port            int
+	PemFile         string
+	KnownHostsFile  string
+	BastionUsername string
+	BastionHost     string
+	BastionPort     int
+	BastionPemFile  string
 }
 
 func (rpla *runnablePlayLocalAnsibleArgs) ToCommandArguments() string {
 	args := fmt.Sprintf("--user='%s'", rpla.Username)
 	if rpla.PemFile != "" {
 		args = fmt.Sprintf("%s --private-key='%s'", args, rpla.PemFile)
 	}
-	args = fmt.Sprintf("%s --ssh-extra-args='-p %d -o UserKnownHostsFile=%s'", args, rpla.Port, rpla.KnownHostsFile)
+
+	sshExtraAgrsOptions := make([]string, 0)
+	sshExtraAgrsOptions = append(sshExtraAgrsOptions, fmt.Sprintf("-p %d", rpla.Port))
+	sshExtraAgrsOptions = append(sshExtraAgrsOptions, fmt.Sprintf("-o UserKnownHostsFile=%s", rpla.KnownHostsFile))
+	if rpla.BastionHost != "" {
+		sshExtraAgrsOptions = append(
+			sshExtraAgrsOptions,
+			fmt.Sprintf(
+				"-o ProxyCommand=\"ssh -p %d -W %%h:%%p %s@%s\"",
+				rpla.BastionPort,
+				rpla.BastionUsername,
+				rpla.BastionHost))
+		if rpla.BastionPemFile == "" {
+			sshExtraAgrsOptions = append(sshExtraAgrsOptions, "-o ForwardAgent=yes")
+			sshExtraAgrsOptions = append(sshExtraAgrsOptions, "-o ConnectTimeout=10")
```

Comment:
Ansible offers a `-T / --timeout` option, currently not implemented but this is what should be used. Implement `-T` and use the value here.